challenger.go

```diff
@@ -260,13 +274,26 @@ func (l *LndChallenger) NewChallenge(price int64) (string, lntypes.Hash, error)
 }
 
 // VerifyInvoiceStatus checks that an invoice identified by a payment
-// hash has the desired status. To make sure we don't fail while the
-// invoice update is still on its way, we try several times until either
-// the desired status is set or the given timeout is reached.
+// hash has the desired status. An optional invoice checker which could
+// be customized by implementer or using the default implementation
+// `DefaultVerifyInvoiceStatus`.
 //
 // NOTE: This is part of the auth.InvoiceChecker interface.
 func (l *LndChallenger) VerifyInvoiceStatus(hash lntypes.Hash,
 	state lnrpc.Invoice_InvoiceState, timeout time.Duration) error {
+	if l.verifyInvoiceStatus != nil {
```

Comment:
Is this check necessary given that `NewChallenger` sets this to non-nil when provided a nil `VerifyInvoiceStatus`?